### PR TITLE
Csra 429 submission api

### PR DESCRIPTION
**Pull request by glenjamin**
Here's a happy path version of the API.

Should be enough here to have the client-side code start calling it.

### CODE REVIEW

File: server/routes/assessment.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  router.post('/', (req, res) => {
+    assessment.record(req.body)
+      .then(
+        result => res.json({
+          status: 'OK',
+          data: { id: result.assessment_id },
+        }),
+        (err) => {
+          // TODO: proper logging lib for server stuff?
```
Comment (romidane) [inline server/routes/assessment.js]: When you do the loging will you add it as a middleware to this router?

Comment (romidane) [inline server/routes/assessment.js]: Can you remove this comment and either capture is as part of ticket?

File: server/routes/assessment.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          // eslint-disable-next-line no-console
+          console.warn('Failed to save assessment', err.stack);
```
Comment (romidane) [inline server/routes/assessment.js]: console.error?

File: test/unit/server/services/assessment.spec.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      getGitDate: sinon.stub(),
+      getQuestionHash: sinon.stub(),
+    };
+
+    assessment = createAssessmentService(fakeDB, fakeAppInfo);
+  }
+
+  describe('risk', () => {
+    describe('validation', () => {
+      it('should have some and be tested');
```
Comment (romidane) [inline test/unit/server/services/assessment.spec.js]: Can we remove it?

File: server/routes/assessment.js (hunk 2 above)
Comment (glenjamin) [inline server/routes/assessment.js]: I tend to use error = crashed app, warn = failed request.

But it makes no different in this scenario as console.warn and console.error both just write to stderr.

File: test/unit/server/services/assessment.spec.js (hunk 3 above)
Comment (glenjamin) [inline test/unit/server/services/assessment.spec.js]: It'll get added in the next commit, didn't see a reason not to leave it in for now.

File: server/routes/assessment.js (hunk 1 above)
Comment (glenjamin) [inline server/routes/assessment.js]: Yeah, proper logging will be configured at the app, and then some middleware that also writes to the main logger per-request.

Comment (glenjamin) [inline server/routes/assessment.js]: I think we should re-work logging in a separate ticket, if I add that jira does it need to go somewhere so it doesn't get lost?

File: test/unit/server/services/assessment.spec.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        expect(fakeDB.returning.lastCall.args[0]).to.eql('assessment_id');
+      });
+      it('resolves with the new ID', () => {
+        expect(result.assessment_id).to.eql(8979);
+      });
+      describe('inserted record', () => {
+        let row;
+        before(() => { row = fakeDB.insert.lastCall.args[0]; });
+
+        it('should set nomis_id from request',
```
Comment (romidane) [inline test/unit/server/services/assessment.spec.js]: not an issue but I generally prefer not using the word "should" in tests. Always felt like it not as assertive as `it('sets a nomis_id from a request')`

Comment (romidane) [inline test/unit/server/services/assessment.spec.js]: would you mind updating them? to match the rest of the test suites?

File: server/routes/assessment.js (hunk 1 above)
Comment (romidane) [inline server/routes/assessment.js]: there is a tech debt epic which it move to

File: test/unit/server/services/assessment.spec.js (hunk 3 above)
Comment (romidane) [inline test/unit/server/services/assessment.spec.js]: right.. as part of this PR?

File: server/routes/assessment.js (hunk 1 above)
Comment (glenjamin) [inline server/routes/assessment.js]: Raised as https://dsdmoj.atlassian.net/browse/CSRA-458 - feel free to move/edit it if that's not the right format/location for this project

File: test/unit/server/services/assessment.spec.js (hunk 3 above)
Comment (glenjamin) [inline test/unit/server/services/assessment.spec.js]: No, but there's no downside? It was part of my original sketch, before I decided how to slice the implementation up into commits.

Will remove it from this commit though, easy enough either way.

**Comment by glenjamin** [comment]
just pushed another commit which updates titles and removes the other placeholder code.

Are you happy with the general approach and structure of the tests / code?

**Pull request merged**